internal/common/actions/create_source_dir/create_source_dir.js

```diff
@@ -0,0 +1,181 @@
+const fs = require("fs-extra");
+const path = require("path");
+const { BazelAction, ensureArray } = require("../run_js/BazelAction");
+
+/**
+ * Action for creating a directory with the passed in files for symlinking and
+ * copying.
+ *
+ * The files can follow a uri syntax with params:
+ * recurse=false (default)
+ * symlink=true (default)
+ *
+ *
+ * The files to be populated have flags set on them for the appropriate actions:
```

Comment:
This is awesome.

Reply:
Thanks! I was trying to make it fairly generic. Would love to hear what you think about the syntax of `flags:some/path:some/other/path:or_a_parameter` I was originally thinking of using URLs with params. But it didn't seem like it would be well suited for specifying associations between multiple paths, which is what this thing needs to do well.

Reply:
I think that should work pretty well. We should always be able to rethink this if need be at a later point.

An alternative would be to use something like a Base64-encoded JSON object or protocol buffer message, removing any issues about special characters and giving us full flexibility in the structure of the object. But that's probably overkill at this point!